File: yetutil/src/main/java/net/yet/ui/dialogs/SpinProgressDlg.java
package net.yet.ui.dialogs;

import android.app.ProgressDialog;
import android.content.Context;

import net.yet.util.TaskUtil;

public class SpinProgressDlg {
	public ProgressDialog dlg;

	public SpinProgressDlg(Context context) {
		dlg = new ProgressDialog(context);
		dlg.setCancelable(true);
		dlg.setCanceledOnTouchOutside(false);
	}

	public SpinProgressDlg setCancelable(boolean cancelable) {
		dlg.setCancelable(cancelable);
		return this;
	}

	public SpinProgressDlg dismiss() {
		dlg.dismiss();
		return this;
	}

	public SpinProgressDlg showLoading() {
		show("加载中...");
		return this;
	}

	public SpinProgressDlg show(String msg) {
		if (dlg.isShowing()) {
			dlg.dismiss();
		}
		dlg.setMessage(msg);
		dlg.show();
		return this;
	}

	public SpinProgressDlg setMsg(final String msg) {
		if (TaskUtil.inMainThread()) {
			dlg.setMessage(msg);
		} else {
			TaskUtil.fore(new Runnable() {
				@Override
				public void run() {
					dlg.setMessage(msg);
				}
			});
		}
		return this;
	}

}

File: yetutil/src/main/java/yet/ui/dialogs/SpinProgressDlg.kt
package yet.ui.dialogs

import android.app.ProgressDialog
import android.content.Context
import yet.util.mainThread

class SpinProgressDlg(context: Context) {
	var dlg: ProgressDialog = ProgressDialog(context)

	init {
		dlg.setCancelable(true)
		dlg.setCanceledOnTouchOutside(false)
	}

	fun cancelable(cancelable: Boolean): SpinProgressDlg {
		mainThread {
			dlg.setCancelable(cancelable)
		}
		return this
	}

	fun dismiss(): SpinProgressDlg {
		mainThread {
			dlg.dismiss()
		}
		return this
	}

	fun showLoading(): SpinProgressDlg {
		show("加载中...")
		return this
	}

	fun show(msg: String): SpinProgressDlg {
		mainThread {
			if (dlg.isShowing) {
				dlg.dismiss()
			}
			dlg.setMessage(msg)
			dlg.show()
		}
		return this
	}

	fun msg(msg: String): SpinProgressDlg {
		mainThread {
			dlg.setMessage(msg)
		}
		return this
	}

}
